packages/core-bridge/bridge-macros/src/lib.rs

```diff
@@ -9,6 +9,9 @@ use proc_macro::TokenStream;
 use syn::{DeriveInput, parse_macro_input};
 
 /// Procedural macro for defining bridge types with compile-time field name conversion
+///
+/// Note that enum types must all be defined on the JS side as objects with a `type` field which
```

Comment:
Ah, yeah, that case is pretty ugly. There were too few of those to be worth optimizing.